etl/steps/data/meadow/oecd/2024-04-30/affordable_housing_database.py

```diff
@@ -0,0 +1,204 @@
+
+    # Number of people experiencing homelessness
+    tb_number = snap.read(sheet_name="Table_HC3.1.A2", skiprows=3, na_values=[".."])
+
+    #
+    # Process data.
+    tb_point_in_time = rename_columns_drop_rows_and_format(
+        tb_point_in_time, columns=COLUMNS_POINT_IN_TIME, short_name="point_in_time", year=False
+    )
+    tb_flow = rename_columns_drop_rows_and_format(tb_flow, columns=COLUMNS_FLOW, short_name="flow", year=False)
+    tb_women = rename_columns_drop_rows_and_format(
+        tb_women, columns=COLUMNS_WOMEN, short_name="share_of_women", year=False
+    )
+
+    # Make tb_index long first
+    tb_index = tb_index.dropna(how="all", ignore_index=True).melt(
+        id_vars=["Unnamed: 11"], var_name="year", value_name="index"
+    )
+    tb_index = rename_columns_drop_rows_and_format(tb_index, columns=COLUMNS_INDEX, short_name="index", year=True)
+
+    # Make tb_trends long first
+    tb_share = tb_share.dropna(how="all", ignore_index=True).melt(
+        id_vars=["Unnamed: 16"], var_name="year", value_name="share"
+    )
+    tb_share = rename_columns_drop_rows_and_format(tb_share, columns=COLUMNS_SHARE, short_name="share", year=True)
+
+    tb_number = remove_unnecesary_rows(tb_number)
+    tb_number = clean_numbers_and_make_long(tb_number)
+    tb_number = rename_columns_drop_rows_and_format(tb_number, columns=COLUMNS_NUMBER, short_name="number", year=True)
+
+    #
+    # Save outputs.
+    #
+    # Create a new meadow dataset with the same metadata as the snapshot.
+    ds_meadow = create_dataset(
+        dest_dir,
+        tables=[
+            tb_point_in_time,
+            tb_flow,
+            tb_women,
+            tb_index,
+            tb_share,
+            tb_number,
+        ],
+        check_variables_metadata=True,
+        default_metadata=snap.metadata,
+    )
+
+    # Save changes in the new meadow dataset.
+    ds_meadow.save()
+
+
+def rename_columns_drop_rows_and_format(tb: Table, columns: Dict[str, str], short_name: str, year: bool) -> Table:
+    """
+    Rename columns, drop nan rows, and format the table.
+    """
+
+    # Assert if columns are in the table
+    assert all(col in tb.columns for col in columns.keys()), log.fatal("Columns not found in the table.")
+
+    # Rename columns
+    tb = tb.rename(columns=columns, errors="raise")
+
+    # Drop nan rows
+    tb = tb.dropna(how="all", ignore_index=True)
+
+    # Drop nan country rows (there are hidden rows in HC3.1.1 sheet)
+    tb = tb.dropna(subset=["country"], how="all", ignore_index=True)
+
+    # Format
+    if year:
+        tb = tb.format(["country", "year"], short_name=short_name)
+    else:
+        tb = tb.format(["country"], short_name=short_name)
+
+    return tb
+
+
+def remove_unnecesary_rows(tb: Table) -> Table:
+    """
+    Remove rows that are not needed.
+    """
+
+    # Drop nan rows in Country
+    tb = tb.dropna(subset=["Country"], how="all", ignore_index=True)
```

Comment:
I don't quite get why I am getting the warning here

Reply:
unsure if related, but isn't the `tb.format` part missing in your meadow code?

Reply:
cc @pabloarosado, in case you know. It's not a big deal, so I would merge anyway.

Reply:
I'm not getting a warning on this line. But I do get one when running `clean_numbers_and_make_long`. To avoid it, you can add `tb = tb.copy()` at the beginning of the function.

Reply:
Thank you! Yeah, the first comment was like 10 days ago. Probably something changed since then. I added that change.